modules/likes/jetpack-likes-settings.php

```diff
@@ -590,22 +590,22 @@ function is_comments_enabled() {
 	function admin_settings_callback() {
 		// We're looking for these, and doing a dance to set some stats and save
 		// them together in array option.
-		$new_state = ! empty( $_POST['wpl_default'] ) ? $_POST['wpl_default'] : 'on';
+		$new_state = ! empty( $_POST['wpl_default'] ) ? $_POST['wpl_default'] : false;
 		$db_state  = $this->is_enabled_sitewide();
 
-		$reblogs_new_state = ! empty( $_POST['jetpack_reblogs_enabled'] ) ? $_POST['jetpack_reblogs_enabled'] : 'on';
+		$reblogs_new_state = ! empty( $_POST['jetpack_reblogs_enabled'] ) ? $_POST['jetpack_reblogs_enabled'] : false;
 		$reblogs_db_state = $this->reblogs_enabled_sitewide();
 		/** Default State *********************************************************/
 
 		// Checked (enabled)
-		switch( $new_state ) {
-			case 'off' :
+		switch ( (bool) $new_state ) {
+			case false:
 				if ( true == $db_state && ! $this->in_jetpack ) {
 					$g_gif = file_get_contents( 'https://pixel.wp.com/g.gif?v=wpcom-no-pv&x_likes=disabled_likes' );
 				}
 				update_option( 'disabled_likes', 1 );
 				break;
-			case 'on'  :
+			case true:
```

Comment:
I'd leave both cases here instead, as well as in the case with `false`. Let's leave it like:
```
case 'on':
case true:
```